src/main/java/org/noureddine/joularjx/monitor/MonitoringHandler.java

```diff
@@ -247,47 +258,91 @@ private Map<Thread, Map<CallTree, Integer>> extractCallTreesStats(Map<Thread, Li
+//	
+//			// If thread already monitored, then calculate CPU time since last time
+//			// Fix for issue #7 (https://github.com/joular/joularjx/issues/7)
+//			// If the last thread cpu time is <= 0, don't subtract, simply take the previous value 
+//			threadCpuTime = threadsCpuTime.merge(entry.getKey().getId(), threadCpuTime,
+//			        (present, newValue) -> newValue - present);
+//			
+//			totalThreadsCpuTime += threadCpuTime;
+//        }
+//        return totalThreadsCpuTime;
+//    }
+//    
+//    /**
+//     * Returns for each thread (PID) it's percentage of CPU time used
+//     * @param threadsCpuTime a map of CPU time per PID
+//     * @param totalThreadsCpuTime the CPU time used by all the monitored threads
+//     * @param processEnergy the energy consumed by the process
+//     * @return for each PID, the percentage of CPU time used by the associated thread
+//     */
+//    private Map<Long, Double> getThreadsCpuTimePercentage(Map<Long, Long> threadsCpuTime,
+//                                                          long totalThreadsCpuTime,
+//                                                          double processEnergy) {
+//        Map<Long, Double> threadsPower = new HashMap<>();
+//        for (var entry : threadsCpuTime.entrySet()) {
+//            double percentageCpuTime = (entry.getValue() * 100.0) / totalThreadsCpuTime;
+//            double threadPower = processEnergy * (percentageCpuTime / 100.0);
+//            threadsPower.put(entry.getKey(), threadPower);
+//        }
+//        return threadsPower;
+//    }
+
+    
     /**
-     * Updates the CPU times for each Thread.
-     * @param methodsStats a map of method occurences for each thread
-     * @param threadsCpuTime a map of CPU time per PID
-     * @return the total CPU time used by all the monitored threads
-     */
-    private long updateThreadsCpuTime(Map<Thread, Map<String, Integer>> methodsStats, Map<Long, Long> threadsCpuTime) {
-        long totalThreadsCpuTime = 0;
-        for (var entry : methodsStats.entrySet()) {
-            long threadCpuTime = threadBean.getThreadCpuTime(entry.getKey().getId());
-
-            threadCpuTime *= entry.getValue().values().stream().mapToDouble(i -> i).sum() / sampleIterations;
-
-            // If thread already monitored, then calculate CPU time since last time
-            threadCpuTime = threadsCpuTime.merge(entry.getKey().getId(), threadCpuTime,
-                    (present, newValue) -> newValue - present);
-
-            totalThreadsCpuTime += threadCpuTime;
-        }
-        return totalThreadsCpuTime;
-    }
-
-    /**
-     * Returns for each thread (PID) it's percentage of CPU time used
-     * @param threadsCpuTime a map of CPU time per PID
-     * @param totalThreadsCpuTime the CPU time used by all the monitored threads
-     * @param processEnergy the energy consumed by the process
-     * @return for each PID, the percentage of CPU time used by the associated thread
+     * 

```

Comment:
Could you add a description of the method in the javadoc? As you merged two methods together, you can just reuse both descriptions into one.